sdcm/utils/quota.py

```diff
@@ -0,0 +1,113 @@
+                                  'GRUB_CMDLINE_LINUX_DEFAULT="quiet splash rootflags=uquota,pquota"/\' ' \
+                                  '/etc/default/grub'
+        mk_grub_cmd = 'grub-mkconfig -o /boot/grub/grub.cfg'
+        enable_uquota_on_mount_cmd = 'sed -i s\'/Options=noatime,discard/Options=noatime,discard,uquota/\' ' \
+                                     '/etc/systemd/system/var-lib-scylla.mount'
+
+        node.remoter.sudo(cmd=change_default_grub_cmd)
+        node.remoter.sudo(cmd=mk_grub_cmd)
+        node.remoter.sudo(cmd=enable_uquota_on_mount_cmd)
+        LOGGER.debug('Rebooting node: "{}"'.format(node.name))
+        node.reboot(hard=False)
+        node.wait_node_fully_start()
+
+
+def verify_quota_is_enabled_on_node(node):
+    """ Verify usrquota is enabled on scylla user. """
+    LOGGER.info("Verifying quota is configured on the node {}".format(node.name))
+    verify_usrquot_in_mount_cmd = "mount | grep /var/lib/scylla | awk {'print $6'} | cut -d ',' -f 8"
+    result = node.remoter.run(cmd=verify_usrquot_in_mount_cmd).stdout.rstrip().replace(")", "")
+    if not result == "usrquota":
+        LOGGER.warning("User quota on user scylla at /var/lib/scylla is not enabled")
+        return False
+    return True
+
+
+def get_currently_used_space_by_scylla_user(node) -> int:
+    result = node.remoter.sudo("xfs_quota -x -c 'report' /var/lib/scylla | grep scylla | tail -1")
+    used_space = int(result.stdout.split()[1])
+    return used_space
+
+
+def configure_quota_on_node_for_scylla_user(node) -> int:
+    """Configure quota with current disk usage size by scylla user + 1GB on /var/lib/scylla mount point"""
+    used_space = get_currently_used_space_by_scylla_user(node)
+    LOGGER.debug('Currently used space by scylla user is: {}K'.format(used_space))
+    quota_size = int(used_space + 1000000)
+    LOGGER.info(
+        "Configuring quota for scylla user with quota size: {}K on node {}".format(quota_size, node))
+    conf_quota_cmd = f"xfs_quota -x -c 'limit bsoft={quota_size}K bhard={quota_size}K scylla' /var/lib/scylla"
+    node.remoter.sudo(conf_quota_cmd, verbose=True, )
+    return quota_size
+
+
+def verify_quota_is_configured_for_scylla_user(node):
+    """Verifies quota ia actually configured for scylla user, but doesn't verify enforcement in ON"""
+    get_scylla_user_quota_value = 'xfs_quota -x -c "report -h" /var/lib/scylla | grep scylla | tail -1 | awk {\'print $4\'}'
+    LOGGER.info("Verifying quota is configured for scylla user")
+    quota_value = node.remoter.sudo(get_scylla_user_quota_value).stdout.rstrip()
+    if quota_value == "0":
+        LOGGER.error("Failed to configure quota on user")
+        raise QuotaConfigurationFailure(
+            "Quota was not configured for scylla user. The value for hard quota is: {}".format(quota_value))
+    LOGGER.info("Quota was configured for scylla user with value: {}".format(quota_value))
+
+
+def remove_quota_configuration_from_node(node):
```

Comment:
this doesn't remove quota configuration from node, rather from user. Fix confusing name.